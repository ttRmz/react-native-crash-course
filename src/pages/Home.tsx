import {NativeStackScreenProps} from '@react-navigation/native-stack';
import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import Animated, {
  useAnimatedStyle,
  withRepeat,
  withSequence,
  withTiming,
} from 'react-native-reanimated';
import {Button} from '../components/Button';
import {useUserContext} from '../contexts/user';
import {NativeStackScreenList} from '../types/routes';

export default function Home({
  navigation,
}: NativeStackScreenProps<NativeStackScreenList, 'home'>) {
  const {logout, user} = useUserContext();
  const animatedStyle = useAnimatedStyle(() => {
    return {
      transform: [
        {
          rotateZ: withSequence(
            withTiming('-10deg', {duration: 50}),
            withRepeat(withTiming('10deg', {duration: 200}), -1, true),
            withTiming('0deg', {duration: 50}),
          ),
        },
      ],
    };
  });

  return (
    <View style={styles.view}>
      <View>
        <Text style={styles.title}>
          Hi {user?.name} !{' '}
          <Animated.View style={animatedStyle}>
            <Text style={styles.emoji}>üëã</Text>
          </Animated.View>
        </Text>
        <Button onPress={() => navigation.push('shows')} style={styles.shows}>
          Browse TV shows üçø
        </Button>
        <Button onPress={() => navigation.push('movies')} style={styles.shows}>
          Browse movies üé¨
        </Button>
      </View>
      <TouchableOpacity onPress={logout} style={styles.logout}>
        <Text style={styles.logoutText}>Logout</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  view: {
    display: 'flex',
    alignItems: 'stretch',
    flexDirection: 'column',
    justifyContent: 'space-between',
    flex: 1,
    marginHorizontal: 44,
    marginTop: 32,
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 24,
  },
  emoji: {
    fontSize: 32,
  },
  shows: {
    marginBottom: 8,
  },
  logout: {
    borderWidth: 2,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 6,
    overflow: 'hidden',
    width: '100%',
    padding: 12,
    marginBottom: 32,
    borderColor: '#b62e2e',
  },
  logoutText: {
    fontWeight: 'bold',
    color: '#b62e2e',
  },
});
