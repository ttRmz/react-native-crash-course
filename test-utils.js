import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';
import React from 'react';

const AllTheProviders = ({children}) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';
// override render method
export {customRender as render};
