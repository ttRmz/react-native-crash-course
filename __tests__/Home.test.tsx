import React from 'react';
import 'react-native';
import Home from '../src/pages/Home';
import {render} from '../test-utils';

test('should render Home page correctly', () => {
  const {toJSON} = render(<Home />);
  expect(toJSON()).toMatchSnapshot();
});
