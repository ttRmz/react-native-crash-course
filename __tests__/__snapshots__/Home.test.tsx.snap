// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home page correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginHorizontal": 44,
      "marginTop": 32,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "fontSize": 32,
          "fontWeight": "bold",
          "justifyContent": "center",
          "marginBottom": 24,
        }
      }
    >
      Hi 
       !
       
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotateZ": 0,
              },
            ],
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          👋
        </Text>
      </View>
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#222",
            "borderRadius": 6,
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 8,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 12,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Browse TV shows 🍿
        </Text>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      handlerTag={2}
      handlerType="NativeViewGestureHandler"
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#222",
            "borderRadius": 6,
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 8,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 12,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Browse movies 🎬
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    handlerTag={3}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#b62e2e",
          "borderRadius": 6,
          "borderWidth": 2,
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": 32,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 12,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#b62e2e",
            "fontWeight": "bold",
          }
        }
      >
        Logout
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
